programs/drift/src/state/lp_pool.rs

```diff
@@ -0,0 +1,1618 @@
+            .safe_mul(in_constituent.gamma_execution as u64)?
+            .safe_div(PERCENTAGE_PRECISION_U64)?
+            .safe_div(2u64)?;
+
+        let out_kappa_execution = out_volatility
+            .safe_mul(out_volatility)?
+            .safe_mul(out_gamma_execution as u64)?
+            .safe_div(PERCENTAGE_PRECISION_U64)?
+            .safe_div(2u64)?;
+
+        // Compute notional targets and errors
+        let in_notional_target = in_target_weight
+            .cast::<i128>()?
+            .safe_mul(self.last_aum.cast::<i128>()?)?
+            .safe_div(PERCENTAGE_PRECISION_I128)?;
+        let in_notional_before =
+            in_constituent.get_notional(in_oracle_price, in_spot_market, 0, true)?;
+        let in_notional_after =
+            in_constituent.get_notional(in_oracle_price, in_spot_market, in_amount, true)?;
+        let in_notional_error_pre = in_notional_before.safe_sub(in_notional_target)?;
+
+        // keep aum fixed if it's a swap for calculating post error, othwerise
+        // increase aum first
+        let in_notional_error_post = if out_spot_market.is_some() {
+            in_notional_after.safe_sub(in_notional_target)?
+        } else {
+            let adjusted_aum = self
+                .last_aum
+                .cast::<i128>()?
+                .safe_add(notional_trade_size)?;
+            let in_notional_target_post_mint_redeem = in_target_weight
+                .cast::<i128>()?
+                .safe_mul(adjusted_aum)?
+                .safe_div(PERCENTAGE_PRECISION_I128)?;
+            in_notional_after.safe_sub(in_notional_target_post_mint_redeem)?
+        };
+
+        let (out_notional_target, out_notional_before, out_notional_after) =
+            if out_constituent.is_some() {
+                (
+                    out_target_weight
+                        .unwrap()
+                        .cast::<i128>()?
+                        .safe_mul(self.last_aum.cast::<i128>()?)?
+                        .safe_div(PERCENTAGE_PRECISION_I128)?,
+                    out_constituent.unwrap().get_notional(
+                        out_oracle_price.unwrap(),
+                        out_spot_market.unwrap(),
+                        0,
+                        true,
+                    )?,
+                    out_constituent.unwrap().get_notional(
+                        out_oracle_price.unwrap(),
+                        out_spot_market.unwrap(),
+                        out_amount.safe_mul(-1)?,
+                        true,
+                    )?,
+                )
+            } else {
+                (0_i128, 0_i128, 0_i128)
+            };
+
+        let out_notional_error_pre = out_notional_before.safe_sub(out_notional_target)?;
+        let out_notional_error_post = out_notional_after.safe_sub(out_notional_target)?;
+
+        // Linear fee computation amount
+        let in_fee_execution_linear = self.get_linear_fee_execution(
+            notional_trade_size,
```

Comment:
seems like the linear and quadratic fee all recalculate `trade size/aum` precentage. Can just calculate it once and pass it down?